packages/wonder-blocks-dropdown/components/action-menu.js

```diff
@@ -191,41 +100,33 @@ export default class ActionMenu extends React.Component<MenuProps, State> {
             children,
             onChange,
             selectedValues,
+            style,
             "aria-disabled": ariaDisabled, // WB-535 avoids passing this prop to the opener
             /* eslint-enable no-unused-vars */
             ...sharedProps
         } = this.props;
-        const {open} = this.state;
-
-        const items = this.getMenuItems();
-
-        const opener = (
-            <ActionMenuOpener
-                {...sharedProps}
-                disabled={items.length === 0 || disabled}
-                onOpenChanged={this.handleOpenChanged}
-                open={open}
-                ref={this.handleOpenerRef}
-                testId={testId}
-            >
-                {menuText}
-            </ActionMenuOpener>
-        );
 
+        const menuItems = React.Children.toArray(this.props.children);
         return (
-            <DropdownCore
+            <Dropdown
                 alignment={alignment}
+                menuItems={menuItems}
+                onChange={this.props.onChange}
+                selectedValues={this.props.selectedValues}
                 dropdownStyle={[styles.menuTopSpace, dropdownStyle]}
-                items={items}
-                keyboard={this.state.keyboard}
-                light={false}
-                onOpenChanged={this.handleOpenChanged}
-                open={open}
-                opener={opener}
-                openerElement={this.openerElement}
-                style={style}
-                role="menu"
-            />
+            >
+                {(eventState) => (
+                    <ActionMenuOpenerCore
+                        {...sharedProps}
+                        {...eventState}
+                        opened={this.state.opened}
+                        disabled={menuItems.length === 0 || disabled}
+                        testId={testId}
```

Comment:
Not sure about this, Is there a way to use `testId` directly into this `ActionMenu` component?

Reply:
I’m not sure what you mean by this?

Reply:
I was referring to using `data-test-id` inside `ActionMenu`, but I just checked and you are doing it inside `ActionMenuOpenerCore` ([here](https://github.com/Khan/wonder-blocks/blob/f5a84c487e3364c08142bed3ed3a34201b08bf18/packages/wonder-blocks-dropdown/components/action-menu-opener-core.js#L84)), and also I'm seeing that `SingleSelect` and `MultiSelect` follow the same pattern.

So please disregard this comment :)